cJSON.h

```diff
@@ -65,6 +65,39 @@ typedef struct cJSON
     char *string;
 } cJSON;
 
+/*
+ * Callback should return non zero if there is an error.
+ */
+typedef int (*cJSON_Callback)(const char *ptr, size_t length, void *cb_data);
+
+/* Forward declare StreamData */
+typedef struct StreamData StreamData;
+
+typedef struct cJSON_Stream
+{
+	/* Stream private data */
+	StreamData *data;
+
+	/* User defined callback function */
+	cJSON_Callback cb;
+	/* User defined data for callback function */
+	void *cb_data;
+
+	/* Whether or not to format the JSON output */
+	int fmt;
```

Comment:
I think this doesn't belong in the stream but should remain a parameter to the print function.